modules/history/src/views/full/MessageTaskBar.tsx

```diff
@@ -0,0 +1,116 @@
+import { Divider, Icon } from '@blueprintjs/core'
+import { AnchorButton, Colors, Position, Tooltip } from '@blueprintjs/core'
+import { Toolbar } from 'botpress/ui'
+import React, { Fragment } from 'react'
+import { CopyToClipboard } from 'react-copy-to-clipboard'
+
+import { MessageGroup } from '../../typings'
+
+import style from './style.scss'
+import { MessageDownload } from './MessageDownload'
+
+interface Filters {
+  flag: boolean
+}
+
+interface Props {
+  currentConv: string
+  updateFilters: (f: Filters) => void
+  useAsFilter: boolean
+  selectedCount: number
+  flag: () => void
+  unflag: () => void
+  messageGroups: MessageGroup[]
+}
+
+interface State {
+  filters: Filters
+  currentConv: string
+  showCopiedTooltip: boolean
+}
+
+export class MessageTaskBar extends React.Component<Props, State> {
+  state = {
+    filters: {
+      flag: false
+    },
+    currentConv: undefined,
+    showCopiedTooltip: false
+  }
+
+  componentDidUpdate() {
+    if (this.props.currentConv !== this.state.currentConv) {
+      this.setState({ currentConv: this.props.currentConv, filters: { flag: false } })
+    }
+  }
+
+  toggleFlagFilter = () => {
+    this.state.filters.flag = !this.state.filters.flag
+    this.props.updateFilters(this.state.filters)
+  }
+
+  renderFilters() {
+    return (
+      <Toolbar>
+        <Fragment>
+          <div className={style.taskBarText}>{this.props.selectedCount} selected messages</div>
```

Comment:
If it doesn't fit in the ButtonGroup of the toolbar, then it should be displayed elsewhere. Also, can you remove the Fragment ? Tooltip/buttons should be first class citizens of the toolbar